appinventor/blocklyeditor/src/component_database.js

```diff
@@ -101,6 +144,12 @@ Blockly.ComponentDatabase = function() {
   /** @type {Object.<string, string>} */
   this.instanceNameUid_ = {};
 
+  /**
+   * Maps names of option lists to OptionLists.
+   * @type {Object.<string, Option>}
+   */
+  this.optionLists_ = {};
```

Comment:
You need to mention in the comment here about how the map in optionsList_ is used to de-duplicate the duplicated optionList info in simpleComponents, so that every option is represented exactly once in the component database, and everything that needs to refer to the optionList does so via the optionList's key.  This field is used in conjunction with the processOptionList method to achieve this behavior.

Reply:
[Done!](https://github.com/BeksOmega/appinventor-sources/pull/4/files#diff-7f92aaa2f89110a614c406da22dc510cR150)